indico/modules/notifications/util.py

```diff
@@ -0,0 +1,68 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2019 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+from functools import wraps
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy.descriptions import RenderMode
+from indico.core.notifications import email_sender
+from indico.modules.notifications import settings as notification_settings
+from indico.modules.notifications.models.notifications import Notification
+
+
+def check_notifications_enabled():
+    url = notification_settings.get('webhook_url')
+    return bool(url)
+
+
+def make_notification(to_users, subject=None, body=None, template=None, content_type='plain_text'):
+    if template is not None and (subject is not None or body is not None):
+        raise ValueError("Only subject/body or template can be passed")
+
+    if template:
+        subject = template.get_subject()
+        body = template.get_body()
+
+    return {
+        'to': set(to_users),
+        'subject': subject,
+        'body': body,
+        'content_type': content_type
+    }
+
+
+def enqueue_notification(notification):
+    for receiver in notification['to']:
+        notification = Notification(
+            user=receiver,
+            subject=notification['subject'],
+            body=notification['body'],
+            render_mode=RenderMode[notification.get('content_type', 'plain_text')]
+        )
+        db.session.add(notification)
```

Comment:
Don't we want to flush here as well?

Reply:
Why? Does it make any difference, since we commit at the end anyway?

Reply:
no, flush is only relevant if you need e.g. the id of a newly inserted object or want to ensure everything has been sent to the DB (e.g. because of other queries depending on it)